packages/@lwc/engine-core/src/framework/template.ts

```diff
@@ -113,6 +115,59 @@ function validateLightDomTemplate(template: Template, vm: VM) {
     }
 }
 
+enum FragmentCache {
+    HAS_TOKEN = 1, // 2^0
+    HAS_SCOPED_STYLE = 2, // 2^1
+    SHADOW_MODE_SYNTHETIC = 4, // 2^2
+}
+
+export function parseFragment(strings: string[], ...keys: number[]): () => Element {
+    const cache = Object.create(null);
+
+    return function (): Element {
+        const {
+            context: { hasScopedStyles, stylesheetToken },
+            shadowMode,
+        } = getVMBeingRendered()!;
+        const hasStyleToken = !isUndefined(stylesheetToken);
+        const isSyntheticShadow = shadowMode === ShadowMode.Synthetic;
+
+        const cacheKey =
+            (hasStyleToken ? FragmentCache.HAS_TOKEN : 0) | // @todo: can this change?
+            (hasScopedStyles ? FragmentCache.HAS_SCOPED_STYLE : 0) | // if styles are reset, we need to recompute.
```

Comment:
This will never change. If `foo.html` has a `foo.scoped.css` file, then it will always have that file. (Unless there is monkey-patching, but let's forget that. 😛 )

Reply:
Nevermind, I see your comment about "if styles are reset." Yes, let's play it safe.